manifests/opsmanager.pp

```diff
@@ -0,0 +1,62 @@
+# This installs Ops Manager. See README.md for more info.
+
+class mongodb::opsmanager (
+  String $user                      = $mongodb::params::user,
+  String $group                     = $mongodb::params::group,
+  Variant[Boolean, String] $ensure  = $mongodb::params::opsmanager_ensure,
+  String $package_name              = $mongodb::params::opsmanager_package_name,
+  Boolean $package_ensure           = $mongodb::params::opsmanager_package_ensure,
+  Boolean $service_enable           = $mongodb::params::opsmanager_service_enable,
+  Boolean $service_manage           = $mongodb::params::opsmanager_service_manage,
+  String $service_name              = $mongodb::params::opsmanager_service_name,
+
+  String $version                 = $mongodb::params::opsmanager_version,
+  String $download_url            = "https://downloads.mongodb.com/on-prem-mms/rpm/mongodb-mms-${version}.x86_64.rpm",
```

Comment:
This should probably be 'optional' or in some way.  As a user I'd want to be able to host the package in a local yum/pulp repo.

Maybe just a `Optional[String[1]] $package_source = undef`?  The README can explain that the package either has to be already available to your package manager but also give an example with the mongodb.com URL being used.

Reply:
The only issue is that Ops Manager isn't currently available in a bonafide repo. The rpms are just hosted in an S3 bucket so as far as I know there is no metadata file to allow yum/pulp access to all available rpms.

Reply:
In my environment, I’d download the rpm and manually upload in to a pulp repo (instead of configuring an upstream to sync from).
I think you’re right that it would be nice if the module did default to something that worked for most people. But for people stuck behind proxies etc. we need to offer a solution.

Making it Optional doesn’t help if you don’t default to `undef`, as you can’t declare a class with a parameter set to `undef` to override the default String.

Maybe an extra parameter is needed. `manage_package_source` or something.

@ekohl you’re normally spot on with suggesting elegant solutions for this sort of thing. Any thoughts? ;)